packages/firestore/src/api/credentials.ts

```diff
@@ -495,10 +496,16 @@ export class FirebaseAppCheckTokenProvider
   private forceRefresh = false;
   private appCheck: FirebaseAppCheckInternal | null = null;
   private latestAppCheckToken: string | null = null;
+  private serverAppAppCheckToken: string | null = null;
 
   constructor(
+    app: FirebaseApp,
     private appCheckProvider: Provider<AppCheckInternalComponentName>
-  ) {}
+  ) {
+    if (_isFirebaseServerApp(app) && app.settings.appCheckToken) {
+      this.serverAppAppCheckToken = app.settings.appCheckToken;
+    }
+  }
```

Comment:
Similiar to the other SDKs (above) during the initialization of the `AppCheckTokenProvider` that irestore implements, check to see if the provided instance of `FirebaseApp` is one of `FirebaseServerApp`, and if it contains an App Check token, then store it locally so it can be returned by the `getToken` method (below).

Reply:
Is there any concern of this token expiring?

Reply:
I attempted to answer this in the main thread of the PR.